apps/spruce/src/pages/hosts/utils.ts

```diff
@@ -7,30 +7,37 @@ import {
 import usePagination from "hooks/usePagination";
 import { useQueryParam } from "hooks/useQueryParam";
 import { mapQueryParamToId } from "types/host";
+import { parseSortString } from "utils/queryString";
 import { HostsQueryParams } from "./constants";
 
 const useQueryVariables = (): HostsQueryVariables => {
   const { limit, page } = usePagination();
   const [currentTaskId] = useQueryParam(HostsQueryParams.CurrentTaskId, "");
   const [distroId] = useQueryParam(HostsQueryParams.DistroId, "");
   const [hostId] = useQueryParam(HostsQueryParams.HostId, "");
-  const [sortBy] = useQueryParam(HostsQueryParams.SortBy, HostSortBy.Status);
-  const [sortDir] = useQueryParam(HostsQueryParams.SortDir, SortDirection.Asc);
+  const [sorts] = useQueryParam(HostsQueryParams.Sorts, "");
   const [startedBy] = useQueryParam(HostsQueryParams.StartedBy, "");
   const [statuses] = useQueryParam(HostsQueryParams.Statuses, []);
 
+  // Parse the sorts query param into an array of sort objects
+  const sortsArray = parseSortString(sorts ?? "", {
+    sortByKey: "sortBy",
+    sortDirKey: "sortDir",
+    sortCategoryEnum: HostSortBy,
+  });
+
+  // Default to status sort if no sorts are provided. We only support one sort at a time.
```

Comment:
I wonder if this can be supported natively by the table. There is a field called `maxMultiSortColCount` that we use in other places, which could help simplify the logic here

Reply:
I found a different command `enableMultiSort` to disable multi sort. Since we are trying to be consistent about the sorts param I think we still need this handling just to handle the array returned by `parseSortString`. But sorry if i'm misunderstanding your comment and you intended something else